pkg/datapath/node.go

```diff
@@ -141,9 +141,16 @@ type NodeNeighbors interface {
 
 type NodeIDHandler interface {
 	// AllocateNodeID allocates a new ID for the given node (by IP) if one wasn't
-	// already assigned.
+	// already assigned or increments the refcnt if it exists.
 	AllocateNodeID(net.IP) uint16
 
+	// DeallocateNodeID decrements the refcnt of a node ID and releases the ID
+	// if the refcnt is zero.
+	DeallocateNodeID(net.IP)
```

Comment:
I have a patch on top of this with unit tests but because we call directly into the bpf map update helpers I had to comment out the bpf map updates to get it to work. This is another reason to move this code up out of datapath/linux into ipcache or elsewhere and build a fake handler to add/delete node ids. I'm trying to minimize the changes here though. I could add the unit test as a patch 3 here though for review, it might help understand the life cycle.